packages/react-meteor-data/ReactMeteorData.jsx

```diff
@@ -1,36 +1,52 @@
-const ReactMeteorData = {
-  componentWillMount() {
-    this.data = {};
-    this._meteorDataManager = new MeteorDataManager(this);
-    const newData = this._meteorDataManager.calculateData();
-    this._meteorDataManager.updateData(newData);
-  },
-  componentWillUpdate(nextProps, nextState) {
-    const saveProps = this.props;
-    const saveState = this.state;
-    let newData;
-    try {
-      // Temporarily assign this.state and this.props,
-      // so that they are seen by getMeteorData!
-      // This is a simulation of how the proposed Observe API
-      // for React will work, which calls observe() after
-      // componentWillUpdate and after props and state are
-      // updated, but before render() is called.
-      // See https://github.com/facebook/react/issues/3398.
-      this.props = nextProps;
-      this.state = nextState;
-      newData = this._meteorDataManager.calculateData();
-    } finally {
-      this.props = saveProps;
-      this.state = saveState;
+import React from 'react';
+
+export function connect({ getMeteorData }, WrappedComponent) {
+  class ReactMeteorData extends React.Component {
```

Comment:
... and sometimes it should not be pure at all, this is why this is optional and set to true per default https://github.com/meteor/react-packages/blob/2e0cbe94b1a8d657559e223006ed0507b212829c/packages/react-meteor-data/createContainer.jsx#L20